### DISCUSSION TITLE
Mark1 FDIM screen configuration

### DISCUSSION THREAD

**Discussion by gokhandeveloper** [jakka351/FG-Falcon#3] [Q&A] [answered]
Is it possible to change the splash screen of FDIM from dual to single zone with Forscan?
I bought a new ICC unit and FDIM is showing dual zone which is fine but I now have a non-functional aircon button for the passenger :).
Thanks.

**Comment by jakka351** [comment]
do you mean the intro boot splash logo or the layout of the air con graphics?

**Reply by gokhandeveloper** [reply]
Layout of the air con graphics. I think the FDIM thinks it is dual unit still and expects control panels at the bottom to be dual zone. I assumed you could program these FDIM units from dual to single or vice versa. 
You won't be able to program the ACM as the hardware is different(premium puts more sound wattage than the base ACM).

**Reply by jakka351** [reply]
For MK1 fdim monochrome screens I dont think you can set it in the options menu of the configuration for FDIM. May be a case of having to manually edit the asbuilt data or load someone else as built file in. The HIM configuration you can set dual and single zone, but that is the HIM side of it, not the display - perhaps that is all that matters though - as the FDIM is displaying graphics depending on what it sees on can message 0x353 - so if the him sends single zone messages it displays single, dual zone messages, displays dual. For the FDIM 7" Mark 1 Colour TFT Screen this will be the case I think, however if you have the monochrome screen - I don't think the graphics are embedded in the screen so it can't actually display them.

I have a monochrome screen that is single zone currently in my car, which is a dual zone set up, also have a dual zone colour FDIM in the car but not plugged in, will try some things and let you know what I find.

Edit: solved this one, found the memory location where the units can be configured for single zone or dual zone. Created software tool to set the zone. Whoopwhoop

**Reply by gokhandeveloper** [reply]
Thanks for your reply.
I had a monochrome screen + single zone. The donor ICC had dual zone colour screen.
So I kept the original ACM+ control panel buttons and changed the FDIMs over only(to have colour screen). 
My control buttons at the bottom is original as well(single zone).
I don't think FDIM looks at the messages from HIM as my HIM is single zone(never had dual zone). It should have changed the graphics to single zone if that was the case?

**Reply by jakka351** [reply]
the FDIM 100% listens for data from the HIM, how else would it know what aircon data to display. the graphics are embedded in the screen and dont change. You dont have enough buttoms, to control the fan speed and passenger temp status. If you set it to single zone it should work fine.

**Reply by gokhandeveloper** [reply]
That's interesting. How did my HIM switch itself to dual then? I never touched or set it to dual.

**Reply by gokhandeveloper** [reply]
Another thing I just realised..
I did not recode the new FDIM to my vehicle's VIN so it may perhaps still be attached to the old VIN which is why HIM isn't able to send messages maybe?
As a result FDIM is stuck in dual zone(last good message from HIM was dual zone).

**Reply by jakka351** [reply]
email me if you are still having issues, can sort it out

**Reply by gokhandeveloper** [reply]
I want to give it a go & program the VIN of my new FDIM to see if it will switch to single zone air con after having the right VIN.
At the moment, I am not able to use the fan speed which is not a deal breaker as the air-con auto functions and the fan speed automatically  adjusts when I +/- the temperature.
I can use the new control panel PCB which has the fan button in the middle but I don't want to. The backlit of the buttons aren't blue like my original :)

I found this scanner which is ms-can compatible and not too expensive.
https://obd2australia.com.au/product/forscan-obd-to-usb-cable-with-ms-can-switch/
What do you think?

**Reply by Bonswally** [reply]
> the FDIM 100% listens for data from the HIM, how else would it know what aircon data to display. the graphics are embedded in the screen and dont change. You dont have enough buttoms, to control the fan speed and passenger temp status. If you set it to single zone it should work fine.

So as far as you know there's no feasible way I could ever buy a dual zone colour fdim and configure it to display the single zone graphics?

They're a fair bit cheaper than the single zone ones is why I'm thinking about it.

**Reply by jakka351** [reply]
You can configure the colour mk 1 FDIM to be single or dual zone, I have software that does it

**Comment by animation303** [comment]
I installed a titanium display in my base model territory, using a combination of titanium fdim titanium switch panel and the original radio unit from my base model and it works perfectly well, the dual zone only displays driver with nothing on the passenger side

**Reply by gokhandeveloper** [reply]
My problem I think is I used my original switch panel. I should have used the new switch panel as well :)

**Reply by jakka351** [reply]
You could use a multimeter to figure out which wires are for which switch, and simply snip and resolder them to change what button does what function.

**Comment by jakka351** [comment, marked as answer]
There is a memory location on the colour MK1 and MK2 screens that you can read via service 0x22 readDataByLocalId and write to via service 0x2E that allows you to set a single zone FDIM to a Dual Zone FDIM and vice versa. I have created a software tool that allows you to convert single zone fdim to dual zone and vice versa - link here https://www.facebook.com/testerPresent%2Fposts%2Fpfbid02DTRSjXhtuYDz3cnkcUnqvWpgyeUGt2Rt2wX6n3gY2gAe9JTxoSXrYJdUDpinJfs4l
